### DISCUSSION TITLE
How to use PinInput with form libs?

### DISCUSSION THREAD

**Discussion by Otoru** [chakra-ui/chakra-ui#3260] [Q&A] [answered]
The vast majority of form libraries work with the idea of using Refs to identify a form's inputs. 

PinInput does not allow us to pass a ref to him (at least we have a TS2322 when we do it) so what is the best way to use it in forms?

I thought about using a hidden component as a contour, but it smells bad. It's the kind of thing that can cause problems in the future

**Comment by with-heart** [comment, marked as answer]
I've never used a form lib that you manage via refs so I'm not sure how you'd expect this to work, but you can pass a ref to individual `PinInputField` elements as each is an `input`.

**Reply by Otoru** [reply]
Yeah!

If someone needs a code reference in the future, look at issue [123](https://github.com/chakra-ui/chakra-ui/issues/3279)
